api/types/provisioning.go

```diff
@@ -1032,22 +1032,16 @@ func (a *ProvisionTokenSpecV2AzureDevops) checkAndSetDefaults() error {
 }
 
 func (a *ProvisionTokenSpecV2BoundKeypair) checkAndSetDefaults() error {
```

Comment:
I'm a bit tempted to make the `bound_keypair` field entirely optional and initialize things here by default.